File: YakBak/app/models/short_words.rb
# frozen_string_literal: true

require 'json'

# Valid English words that are 1-2 characters long.
# Words shorter than 3 characters must appear in this set to be accepted in submissions.
# Sourced from shared/short_words.json (single source of truth for frontend + backend).
module ShortWords
  SHARED_PATH = File.expand_path('../../shared/short_words.json', __dir__)

  VALID = Set.new(JSON.parse(File.read(SHARED_PATH))).freeze

  def self.valid_word_length?(word)
    return true if word.length >= 3

    VALID.include?(word.downcase)
  end
end

File: Yak/src/constants/shortWords.ts
import shortWordsList from "../../shared/short_words.json";

/**
 * Valid English words that are 1-2 characters long.
 * Words shorter than 3 characters must appear in this set to be accepted in submissions.
 * Sourced from shared/short_words.json (single source of truth for frontend + backend).
 */
export const SHORT_WORDS: Set<string> = new Set(shortWordsList);

/**
 * Check if a word meets the minimum length requirement.
 * Words must be 3+ characters, or be in the short words allowlist.
 */
export function isValidWordLength(word: string): boolean {
  if (word.length >= 3) return true;
  return SHORT_WORDS.has(word.toLowerCase());
}
